thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -0,0 +1,608 @@
+          for (String dimensionValue : dimensionValues) {
+            StringBuffer sb = new StringBuffer();
+            dimensionsArray[dimensionIndex] = dimensionValue;
+            sb.append(Joiner.on(DIMENSION_JOINER).join(Lists.newArrayList(dimensionsArray)));
+            dimensionPatternsList.add(sb.toString());
+            dimensionsArray[dimensionIndex] = STAR_DIMENSION;
+          }
+        }
+        dimensionPatterns = new String[dimensionPatternsList.size()];
+        dimensionPatterns = dimensionPatternsList.toArray(dimensionPatterns);
+      }
+
+      if (StringUtils.isNotBlank(metric)) {
+        if (StringUtils.isNotBlank(dimensions)) {
+          anomalyResults = anomalyResultDAO.findAllByCollectionTimeMetricAndDimensions(dataset, metric,
+              startTime, endTime, dimensionPatterns);
+        } else {
+          anomalyResults = anomalyResultDAO.findAllByCollectionTimeAndMetric(dataset, metric,
+              startTime, endTime);
+        }
+      } else {
+        anomalyResults = anomalyResultDAO.findAllByCollectionAndTime(dataset, startTime, endTime);
+      }
+
+    } catch (Exception e) {
+      LOG.error("Exception in fetching anomalies", e);
+    }
+    return anomalyResults;
+  }
+
+  /************* CRUD for anomaly functions of collection **********************************************/
+
+  // View all anomaly functions
+  @GET
+  @UnitOfWork
+  @Path("/anomaly-function/view")
+  public List<AnomalyFunctionSpec> viewAnomalyFunctions(@QueryParam("dataset") String dataset, @QueryParam("metric") String metric) {
+
+    List<AnomalyFunctionSpec> anomalyFunctionSpecs = anomalyFunctionSpecDAO.findAllByCollection(dataset);
+    List<AnomalyFunctionSpec> anomalyFunctions = anomalyFunctionSpecs;
+
+    if (StringUtils.isNotEmpty(metric)) {
+      anomalyFunctions = new ArrayList<>();
+      for (AnomalyFunctionSpec anomalyFunctionSpec : anomalyFunctionSpecs) {
+        if (metric.equals(anomalyFunctionSpec.getMetric())) {
+          anomalyFunctions.add(anomalyFunctionSpec);
+        }
+      }
+    }
+    return anomalyFunctions;
+  }
+
+  // Add anomaly function
+  @POST
+  @UnitOfWork
+  @Path("/anomaly-function/create")
+  public Response createAnomalyFunction(@QueryParam("dataset") String dataset,
+      @QueryParam("functionName") String functionName,
+      @QueryParam("metric") String metric,
+      @QueryParam("type") String type,
+      @QueryParam("windowSize") String windowSize,
+      @QueryParam("windowUnit") String windowUnit,
+      @QueryParam("windowDelay") String windowDelay,
+      @QueryParam("windowDelayUnit") String windowDelayUnit,
+      @QueryParam("scheduleMinute") String scheduleMinute,
+      @QueryParam("scheduleHour") String scheduleHour,
+      @QueryParam("repeatEvery") String repeatEvery,
+      @QueryParam("exploreDimension") String exploreDimensions,
+      @QueryParam("properties") String properties,
+      @QueryParam("isActive") boolean isActive)
```

Comment:
add comment explaining that bucketSize, bucketUnit are defaulted to collection granularity.
Add filters param.